### PR TITLE
Fix #152 Add imagePullSecrets to deployments with a global default

### LINKED ISSUE #152: Add imagePullSecrets and namespaces to helm template
You can see the example below for imagePullSecrets and the same could be done for namespaces.

```
spec:
      {{- if .Values.image.pullSecret -}}
      imagePullSecrets:
        - name: {{ .Values.image.pullSecret }}
      {{- end }}
```

This will help companies deploy harbor using helm + private internal repository and/or scope it to a namespace.

### PR DESCRIPTION
**Pull request by seyhbold**
This PR adds a global default:

```
imagePullSecrets: []
```

and an `imagePullSecrets:` for every deployment.
to `values.yaml`

The template first look in the specific entry in `values.yaml` and default to the global entry which is an empty array by default.

Fixes #152

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -224,6 +224,11 @@ persistence:
 
 imagePullPolicy: IfNotPresent
 
+# Use this set to assign a list of default pullSecrets
+imagePullSecrets: []
```
Comment (ywk253100) [inline values.yaml]: I think we can remove the `[]` at the end of this line as users need to do that if they want to add a secret

File: values.yaml [hunk 2]
```diff
@@ -235,6 +240,9 @@ nginx:
   image:
     repository: goharbor/nginx-photon
     tag: dev
+    # Override default imagePullSecrets
```
Comment (ywk253100) [inline values.yaml]: Is it necessary to provide a secret for each image? In what scenario this is needed?

File: values.yaml (hunk 1 above)
Comment (seyhbold) [reply values.yaml]: You're right. I'll remoe it.
Should I leave it as a comment?

File: values.yaml (hunk 2 above)
Comment (seyhbold) [reply values.yaml]: I am using public registries like hub.docker.com or quay.io along with a locally hosted for my own developments.
I do not want to see the list of all secrets in every deployment, but this is more to my own taste.
It is definitely not necessary.
Do you want it removed?

File: values.yaml (hunk 1 above)
Comment (seyhbold) [reply values.yaml]: Ah, I should have read more thoroughly. I'll remove the `[]`

File: values.yaml (hunk 2 above)
Comment (ywk253100) [reply values.yaml]: I think that keep only one overall config is better.

**Comment by ywk253100** [comment]
Add DCO is needed for you PR.

File: README.md [hunk 3]
```diff
@@ -145,12 +145,14 @@ The following table lists the configurable parameters of the Harbor chart and th
...
 | `nginx.image.tag`                                                           | Image tag                                                                                                                                                                                                                                                                                                                                       | `dev`                           |
+| `nginx.image.imagePullSecrets`                                              | Add imagePullSecret names to the nginx deployment                                                                                                                                                                                                                                                                                               |                                 |
```
Comment (ywk253100) [inline README.md]: I think these items are not needed anymore either

**Comment by seyhbold** [comment]
Uhh sorry, I forgot.

File: README.md (hunk 3 above)
Comment (seyhbold) [reply README.md]: Uhh sorry, I forgot.

**Comment by ywk253100** [comment]
@seyhbold Thanks

**Pull request merged**